### PR TITLE
Add mailbox command for DDR Generic capture.

### PR DESCRIPTION
**Pull request by nishitpatel28**

### CODE REVIEW

**Review by nishitpatel28 — COMMENTED** [review]
**ddr_generic_capture output:**
[*********]# ./cxl/cxl perfcnt-ddr-generic-capture mem0
in size: 0x8
command completed successfully
=========================== PERFCNT DDR Generic Capture ============================
Generic Counter Readings:
0
0
0
0
0
0
0
0

**ddr_dfi_capture output:**
[*********]# ./cxl/cxl perfcnt-ddr-dfi-capture mem0
in size: 0x8
command completed successfully
=========================== PERFCNT DDR DFI Capture ============================
DFI Counter Readings:
DFI Counter 17: 0
DFI Counter 20: 0
DFI Counter 21: 0
DFI CH1 Counter 17: 0
DFI CH1 Counter 20: 0
DFI CH1 Counter 21: 0

File: cxl/lib/libcxl.c [hunk 1]
```diff
@@ -6265,3 +6265,174 @@ CXL_EXPORT int cxl_memdev_err_inj_hif_ecc(struct cxl_memdev *memdev,
...
+		if (!cmd->input_payload)
+			return -ENOMEM;
+		cmd->send_cmd->in.payload = (u64)cmd->input_payload;
+		cmd->send_cmd->in.size = cinfo->size_in;
+	}
+	fprintf(stdout, "in size: 0x%x\n", cmd->send_cmd->in.size);
+
+	perfcnt_ddr_generic_capture_in = (void *) cmd->send_cmd->in.payload;
+	perfcnt_ddr_generic_capture_in->ddr_id = ddr_id;
+	perfcnt_ddr_generic_capture_in->poll_period_ms = poll_period_ms;
```
Comment (elake) [inline cxl/lib/libcxl.c]:
```suggestion
	perfcnt_ddr_generic_capture_in->poll_period_ms = cpu_to_le32(poll_period_ms);
```

File: cxl/lib/libcxl.c [hunk 2]
```diff
@@ -6265,3 +6265,174 @@ CXL_EXPORT int cxl_memdev_err_inj_hif_ecc(struct cxl_memdev *memdev,
...
+
+#define CXL_MEM_COMMAND_ID_PERFCNT_DDR_GENERIC_CAPTURE CXL_MEM_COMMAND_ID_RAW
+#define CXL_MEM_COMMAND_ID_PERFCNT_DDR_GENERIC_CAPTURE_OPCODE 0XCA11
+#define CXL_MEM_COMMAND_ID_PERFCNT_DDR_GENERIC_CAPTURE_PAYLOAD_IN_SIZE 8
+#define CXL_MEM_COMMAND_ID_PERFCNT_DDR_GENERIC_CAPTURE_PAYLOAD_OUT_SIZE 32
+
+struct cxl_mbox_perfcnt_ddr_generic_capture_in {
+	u8 ddr_id;
+	u8 rsvd[3];
+	u32 poll_period_ms;
```
Comment (elake) [inline cxl/lib/libcxl.c]:
```suggestion
	__le32 poll_period_ms;
```

File: cxl/lib/libcxl.c [hunk 3]
```diff
@@ -6265,3 +6265,174 @@ CXL_EXPORT int cxl_memdev_err_inj_hif_ecc(struct cxl_memdev *memdev,
...
+
+#define CXL_MEM_COMMAND_ID_PERFCNT_DDR_DFI_CAPTURE CXL_MEM_COMMAND_ID_RAW
+#define CXL_MEM_COMMAND_ID_PERFCNT_DDR_DFI_CAPTURE_OPCODE 0XCA12
+#define CXL_MEM_COMMAND_ID_PERFCNT_DDR_DFI_CAPTURE_PAYLOAD_IN_SIZE 8
+#define CXL_MEM_COMMAND_ID_PERFCNT_DDR_DFI_CAPTURE_PAYLOAD_OUT_SIZE 24
+
+struct cxl_mbox_perfcnt_ddr_dfi_capture_in {
+	u8 ddr_id;
+	u8 rsvd[3];
+	u32 poll_period_ms;
```
Comment (elake) [inline cxl/lib/libcxl.c]:
```suggestion
	__le32 poll_period_ms;
```

File: cxl/lib/libcxl.c [hunk 4]
```diff
@@ -6265,3 +6265,174 @@ CXL_EXPORT int cxl_memdev_err_inj_hif_ecc(struct cxl_memdev *memdev,
...
+		if (!cmd->input_payload)
+			return -ENOMEM;
+		cmd->send_cmd->in.payload = (u64)cmd->input_payload;
+		cmd->send_cmd->in.size = cinfo->size_in;
+	}
+	fprintf(stdout, "in size: 0x%x\n", cmd->send_cmd->in.size);
+
+	perfcnt_ddr_dfi_capture_in = (void *) cmd->send_cmd->in.payload;
+	perfcnt_ddr_dfi_capture_in->ddr_id = ddr_id;
+	perfcnt_ddr_dfi_capture_in->poll_period_ms = poll_period_ms;
```
Comment (elake) [inline cxl/lib/libcxl.c]:
```suggestion
	perfcnt_ddr_dfi_capture_in->poll_period_ms = cpu_to_le32(poll_period_ms);
```

**Review by elake — CHANGES_REQUESTED** [review]

**Review by elake — APPROVED** [review]

**Pull request merged**